File: Heat/Models/Plant.vb
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Namespace Models
    ''' <summary>
    ''' Rappresenta un Impianto: un impianto ha un libretto di impianto.
    ''' </summary>
    ''' <remarks></remarks>
    Public Class Plant

        Sub New()
            Me.Contacts = New List(Of Contact)
        End Sub

        <Key> _
        Property ID As Integer

        <Display(name:="Codice impianto")> _
        Property Code As Integer
        <Display(name:="Nominativo")> _
        Property Name As String

        <Display(name:="Classe impianto")> _
        Property PlantClass As PlantClass

        <Display(name:="Tipologia impianto")> _
        Property PlantType As PlantType

        <Display(name:="Codice dell'impianto per la provincia")> _
        Property PlantDistinctCode As String

        Property BuildingAddress As PlantBuilding

        Property ThermalUnit As ThermalUnit

        Property Contacts As List(Of Contact)

        Overridable Property Service As PlantService

    End Class

End Namespace

File: Heat.ConvertedToC#/Models/Plant.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Heat.Models
{
    /// <summary>
    /// Rappresenta un Impianto: un impianto ha un libretto di impianto.
    /// </summary>
    /// <remarks></remarks>
    public class Plant
	{

		public Plant()
		{
			this.Contacts = new List<Contact>();
			this.Media = new List<Medium>();
		}

		[Key()]
		public int ID { get; set; }

		[Display(Name = "Codice impianto")]
		public int Code { get; set; }
		[Display(Name = "Nominativo")]
		public string Name { get; set; }

		[Display(Name = "Classe impianto")]
		public PlantClass PlantClass { get; set; }

		[Display(Name = "Tipologia impianto")]
		public PlantType PlantType { get; set; }

		[Display(Name = "Codice dell'impianto per la provincia")]
		public string PlantDistinctCode { get; set; }

		public PlantBuilding BuildingAddress { get; set; }

		public ThermalUnit ThermalUnit { get; set; }

		public List<Contact> Contacts { get; set; }

		public PlantService Service { get; set; }

		public List<Medium> Media { get; set; }

	}

}
